js/inputbar.js

```diff
@@ -100,6 +100,14 @@ weechat.directive('inputBar', function() {
                         ab.clear();
                     }
 
+                    // Cache our queries so when a buffer gets opened we can open in UI
+                    if ($scope.command.substring(0, 7) === '/query ') {
```

Comment:
This logic seems unnecessarily fragile, why not something along the lines of
```js
var command = $scope.command.split(' ').splice(0, 1); // or /\s/
var commands = ['/query', '/q'];
var index = commands.indexOf(command);
if (index >= 0) {
    var queryName = $scope.command.substring(commands[index].length + 1);
    models.outgoingQueries.push(queryName);
}
```
that would be easily extendable to '/join' as well.